### PR TITLE
nixos/mbpfan: revise defaults

### PR DESCRIPTION
**Pull request by bb2020**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/mbpfan.nix [hunk 1]
```diff
@@ -24,10 +24,10 @@ in {
       description = lib.mdDoc "If true, sets the log level to verbose.";
     };
 
-    aggressive = mkOption {
```
Comment (aanderse) [inline nixos/modules/services/misc/mbpfan.nix:27]: Can we please add a `mkRemovedOptionModule` here.

File: nixos/modules/services/misc/mbpfan.nix [hunk 2]
```diff
@@ -70,13 +73,6 @@ in {
   ];
 
   config = mkIf cfg.enable {
-    services.mbpfan.settings = mkIf cfg.aggressive {
```
Comment (aanderse) [inline nixos/modules/services/misc/mbpfan.nix:73]: Might be a good idea to mention in release notes that defaults have changed. Maybe include a justification?

**Review by aanderse — CHANGES_REQUESTED** [review]

File: nixos/modules/services/misc/mbpfan.nix (hunk 1 above)
Comment (bb2020) [reply nixos/modules/services/misc/mbpfan.nix:27]: Maybe I should just enable this by default, not to break compatibility?

**Comment by bb2020** [comment]
@ofborg eval

File: nixos/modules/services/misc/mbpfan.nix (hunk 1 above)
Comment (aanderse) [reply nixos/modules/services/misc/mbpfan.nix:27]: You mean not remove the option?

Comment (bb2020) [reply nixos/modules/services/misc/mbpfan.nix:27]: Yes. I've made some changes, can you review again?

File: nixos/modules/services/misc/mbpfan.nix (hunk 2 above)
Comment (aanderse) [reply nixos/modules/services/misc/mbpfan.nix:73]: Release notes mentioning changes?

Comment (bb2020) [reply nixos/modules/services/misc/mbpfan.nix:73]: Done

**Review by aanderse — APPROVED** [review]
thanks

**Pull request merged**